### PR TITLE
*: use AllSchemaNames to replace AllSchemas

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: ref #50959

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [x] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/infoschema/infoschema.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ type InfoSchema interface {
 	SchemaByID(id int64) (*model.DBInfo, bool)
 	TableByID(id int64) (table.Table, bool)
 	AllSchemas() []*model.DBInfo
```
Comment (tiancaiamao) [inline pkg/infoschema/infoschema.go:45]: Can we remove this API now?

Comment (wjhuang2016) [reply pkg/infoschema/infoschema.go:45]: Still have some usages. I will remove them after this PR.

File: pkg/infoschema/infoschema_v2.go [hunk 2]
```diff
@@ -343,6 +343,15 @@ func (is *infoschemaV2) AllSchemas() (schemas []*model.DBInfo) {
 	return
 }
 
+func (is *infoschemaV2) AllSchemaNames() []model.CIStr {
+	rs := make([]model.CIStr, 0, is.Data.schemaMap.Len())
```
Comment (tiancaiamao) [inline pkg/infoschema/infoschema_v2.go:347]: You need to handle `is.Data.specials` like here https://github.com/pingcap/tidb/pull/51436/files#diff-d82e5bb6499a8d8c00f3d55ec95bccda3fab6d4cc45529cecf06552bfa973b1bL352-L355

File: pkg/infoschema/infoschema_v2.go [hunk 3]
```diff
@@ -343,6 +343,18 @@ func (is *infoschemaV2) AllSchemas() (schemas []*model.DBInfo) {
...
+	is.Data.schemaMap.Scan(func(item schemaItem) bool {
+		rs = append(rs, item.dbInfo.Name)
+		return true
+	})
```
Comment (ywqzzy) [inline pkg/infoschema/infoschema_v2.go:351]: Is it safe for us only fetch schema from cache instead of tikv?

Comment (wjhuang2016) [reply pkg/infoschema/infoschema_v2.go:351]: Same as AllSchemas()

**Review by ywqzzy — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Pull request merged**